src/elm/Page/Device.elm

```diff
@@ -220,6 +292,33 @@ handleModalCommand session model cmd =
             Cmd.none
 
 
+handleGroupModalCommand : Session -> Model -> SelectGroup.ExternalMsg -> Cmd Msg
+handleGroupModalCommand session model cmd =
+    case ( model.device, cmd ) of
+        ( Just device, SelectGroup.SelectedGroup groupName ) ->
+            AstarteApi.addDeviceToGroup session.apiConfig groupName model.deviceId AddGroupToDeviceDone
+
+        ( Nothing, _ ) ->
+            Cmd.none
+
+        ( _, SelectGroup.Noop ) ->
+            Cmd.none
+
+
+subtractList : List a -> List a -> List a
+subtractList listA listB =
+    List.foldl
+        (\item acc ->
+            if List.member item listB then
```

Comment:
You should probably convert listB to a `Set` since you're constantly looking it up for membership, otherwise you have to go through all listB for every failing lookup.

You can also convert both lists to `Set` and `diff` them, but you would lose the ordering (I'm not sure if it matters for you here)